tests/conftest.py

```diff
@@ -3110,3 +3110,134 @@ def finalizer():
 
     request.addfinalizer(finalizer)
     return factory
+
+
+@pytest.fixture(scope='function')
+def multiple_snapshot_and_clone_of_postgres_pvc_factory(
+    request, pgsql_factory_fixture, snapshot_factory,
+    snapshot_restore_factory, pvc_clone_factory
+):
+    """
+    Calling this fixture creates multiple snapshots & clone of postgres PVC
+    """
+    instances = []
+
+    def factory(pvc_size_new, pgsql):
+        """
+        Args:
+            pvc_size_new (int): Resize/Expand the pvc size
+            pgsql (obj): Pgsql obj
+
+        Returns:
+            Postgres pod: Pod instances
+
+        """
+
+        # Get postgres pvc list obj
+        postgres_pvcs_obj = pgsql.get_postgres_pvc()
+
+        # Take a snapshot
```

Comment:
there are similarities between the implementation here and in a few other places:
https://github.com/red-hat-storage/ocs-ci/blob/4723cef0d34a27ab287cb6c4900227c1cb9fb59a/tests/e2e/lifecycle/pvc_snapshot/test_pgsql_pvc_snapshot.py
https://github.com/red-hat-storage/ocs-ci/blob/0bd34b15267e4b993682e44a576957ab2157ce24/tests/manage/pv_services/pvc_snapshot/test_snapshot_at_different_pvc_utlilization_level.py
https://github.com/red-hat-storage/ocs-ci/blob/ecac62e12ab3f0f3330eb191d406ea1b8d2f198f/tests/manage/pv_services/pvc_snapshot/test_rbd_block_pvc_snapshot.py


Will it be possible to merge some of them to function/s to avoid duplicity?

Reply:
Agreed, added function to create multiple snapshots, clones on confest.py and using those.Will send another PR for using these functions.